mlrun/api/db/sqldb/db.py

```diff
@@ -3460,3 +3524,104 @@ def _is_background_task_timeout_exceeded(background_task_record) -> bool:
+            for notification in self._find_notifications(session, run_id=run.id)
+        }
+        notifications = []
+        for notification_model in notification_objects:
+            new_notification = False
+            notification = run_notifications.get(notification_model.name, None)
+            if not notification:
+                new_notification = True
+                notification = Notification(
+                    name=notification_model.name, run=run.id, project=project
+                )
+
+            notification.kind = notification_model.kind
+            notification.message = notification_model.message
+            notification.severity = notification_model.severity
+            notification.when = ",".join(notification_model.when)
+            notification.condition = notification_model.condition
+            notification.params = notification_model.params
+            notification.status = (
+                notification_model.status
+                or mlrun.api.schemas.NotificationStatus.PENDING
+            )
+            notification.sent_time = notification_model.sent_time
+
+            logger.debug(
+                f"Storing {'new' if new_notification else 'existing'} notification",
+                notification_name=notification.name,
+                run_uid=run_uid,
+                project=project,
+            )
+            notifications.append(notification)
+
+        self._upsert(session, notifications)
+
+    def list_notifications(
+        self,
+        session,
+        run_uid: str,
+        project: str = "",
+    ) -> typing.List[mlrun.model.Notification]:
+        run = self._get_run(session, run_uid, project, 0)
+        if not run:
+            return []
+
+        return [
+            self._transform_notification_record_to_schema(notification)
+            for notification in self._query(session, Notification, run=run.id).all()
+        ]
+
+    def delete_notifications(
+        self,
+        session,
+        name: str = None,
+        run_uid: str = None,
+        project: str = None,
+        commit: bool = True,
+    ):
+        run_id = None
+        if run_uid:
+            run = self._get_run(session, run_uid, project, 0)
+            if not run:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Run not found: uid={run_uid}, project={project}"
+                )
+            run_id = run.id
+
+        project = project or config.default_project
+        if project == "*":
+            project = None
+
+        k8s = mlrun.api.utils.singletons.k8s.get_k8s()
+
+        query = self._find_notifications(session, name, run_id, project)
+        for notification in query:
+            if (
+                notification.params
+                and "secret" in notification.params
+                and k8s.running_inside_kubernetes_cluster
+            ):
+                k8s.delete_secrets(notification.params["secret"], None)
```

Comment:
k8s secret handling should not be done here. The `crud` object should handle it.